appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/stateControl/registry/impl/KogitoCommandRegistry.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.stateControl.registry.impl;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.appformer.kogito.api.stateControl.registry.CommandRegistry;
+import org.appformer.kogito.api.stateControl.registry.RegistryChangeListener;
+import org.appformer.kogito.bridge.client.interop.WindowRef;
+import org.appformer.kogito.bridge.client.stateControl.interop.StateControl;
+import org.appformer.kogito.bridge.client.stateControl.registry.interop.KogitoJSCommandRegistry;
+
+public class KogitoCommandRegistry<C> implements CommandRegistry<C> {
```

Comment:
Could you please explain why **appformer-kogito-bridge** module needs to have this **KogitoCommandRegistry** registry implementation? **KogitoCommandRegistry** seems same as **DefaultCommandRegistryImpl** from **appformer-kogito-api**. I expected in **appformer-kogito-bridge** just **KogitoJSCommandRegistry**.

Reply:
Well, the `KogitoCommandRegistry` is the CDI bean that will be injected instead of the `DefaultCommandRegistry` when using the envelope. It basically wraps the JS registry from the envelope, since the JS cannot be directly injected. That's the main point of having that `KogitoCommandRegistry`.